libs/s25main/AddonHelperFunctions.h

```diff
@@ -0,0 +1,11 @@
+// Copyright (C) 2005 - 2025 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "GamePlayer.h"
```

Comment:
```suggestion
#include <functional>

class GamePlayer;
```

And we use west-const (i.e. `const GamePlayer&`) below (IIRC we could upgrade clang-format to do that automatically)

No need to have enums as references (GoodType, Job)

Reply:
Use correct const order and remove reference for trivial types